packages/gateway/src/features/registration/type-resolvers.ts

```diff
@@ -1241,6 +1243,66 @@ export const typeResolvers: GQLResolver = {
       }
 
       return false
+    },
+    async certifier(
+      parent: { docRef: DocumentReference; task: Saved<Task> },
+      _,
+      context
+    ) {
+      const { docRef, task } = parent
+      const practitionerRef =
+        docRef.extension &&
+        findExtension(
+          `${OPENCRVS_SPECIFICATION_URL}extension/collector`,
+          docRef.extension
+        )
+
+      if (
+        practitionerRef &&
+        practitionerRef.valueReference &&
+        practitionerRef.valueReference.reference &&
+        practitionerRef.valueReference.reference.startsWith('Practitioner')
+      ) {
+        const practitionerId = resourceIdentifierToUUID(
+          practitionerRef.valueReference.reference as ResourceIdentifier
+        )
+        const practitionerRoleBundle =
+          await context.dataSources.fhirAPI.getPractitionerRoleByPractitionerId(
+            practitionerId
+          )
+
+        const practitionerRoleId =
+          practitionerRoleBundle.entry?.[0].resource?.id
+
+        const practitionerRoleHistory =
+          await context.dataSources.fhirAPI.getPractionerRoleHistory(
+            practitionerRoleId
+          )
+        const result = practitionerRoleHistory.find(
+          (it) =>
+            it?.meta?.lastUpdated &&
+            task.lastModified &&
+            it?.meta?.lastUpdated <= task.lastModified!
+        )
+
+        const targetCode = result?.code?.find((element) => {
+          return (
+            element.coding?.[0].system === 'http://opencrvs.org/specs/roles'
+          )
+        })
+
+        const roleId = targetCode?.coding?.[0].code
+        const userResponse =
+          await context.dataSources.usersAPI.getUserByPractitionerId(
+            practitionerId
+          )
+
+        const allRoles = await context.dataSources.countryConfigAPI.getRoles()
+        const role = allRoles.find((role) => role.id === roleId)?.id
+
+        return { ...userResponse, role }
+      }
+      return null
```

Comment:
Let's do a early return here to make it a bit easier to follow, e.g.
```
if (!practitionerRef?.valueReference?.reference?.startsWith('Practitioner')) {
  return null
}
```

Reply:
Also could we re-use this function: https://github.com/opencrvs/opencrvs-core/blob/c2fe1cd66bc1aaab6b0a4b1c1ea366af77560cbd/packages/gateway/src/features/registration/type-resolvers.ts#L1505? Or update it accordingly?

Reply:
Not sure if we can do the 2nd one. The use of `getUserRoleFromHistory` was removed on [this PR](https://github.com/opencrvs/opencrvs-core/pull/7301/files#). And this is how it currently is as well. You might check `packages/gateway/src/features/registration/type-resolvers.ts` inside it.